server/game/game.js

```diff
@@ -22,7 +22,25 @@ class Game extends EventEmitter {
     }
 
     addMessage(message) {
-        this.messages.push({ date: new Date(), message: message });
+        var args = Array.from(arguments).slice(1);
+        var formattedMessage = this.formatMessage(message, args);
+        this.messages.push({ date: new Date(), message: formattedMessage });
+    }
+
+    formatMessage(format, args) {
+        var messageFragments = format.split(/(\{\d+\})/);
+        return _.map(messageFragments, fragment => {
```

Comment:
This whole method is still pretty ugly but at least it's server side now. Client is only responsible for rendering what the server tells it now.